src/box/lua/net_box.c

```diff
@@ -1895,7 +2010,10 @@ luaT_netbox_request_wait_result(struct lua_State *L)
 			return luaT_push_nil_and_error(L);
 		}
 	}
-	return netbox_request_push_result(request, L);
+	int rc = netbox_request_push_result(request, L);
+	if (rc == 1)
+		rc = netbox_request_prepare_result(request, L);
```

Comment:
Can you do this right in `netbox_request_push_result`?

Reply:
Then the change will affect non-future requests

Reply:
Discussed f2f, decided to return table if `fetch_pos` is true.